specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json

```diff
@@ -0,0 +1,1463 @@
+        },
+        "vnetSubnetID": {
+          "$ref": "#/definitions/ContainerServiceVnetSubnetID",
+          "description": "VNet SubnetID specifies the vnet's subnet identifier."
+        },
+        "firstConsecutiveStaticIP": {
+          "type": "string",
+          "description": "FirstConsecutiveStaticIP used to specify the first static ip of masters.",
+          "default": "10.240.255.5"
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/ContainerServiceStorageProfile",
+          "description": "Storage profile specifies what kind of storage used. Choose from StorageAccount and ManagedDisks. Leave it empty, we will choose for you based on the orchestrator choice."
+        },
+        "fqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "FDQN for the master pool."
+        }
+      },
+      "required": [
+        "dnsPrefix",
+        "vmSize"
+      ],
+      "description": "Profile for the container service master."
+    },
+    "ManagedClusterAgentPoolProfile": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique name of the agent pool profile in the context of the subscription and resource group."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "maximum": 100,
+          "minimum": 1,
+          "description": "Number of agents (VMs) to host docker containers. Allowed values must be in the range of 1 to 100 (inclusive). The default value is 1. ",
+          "default": 1
+        },
+        "vmSize": {
+          "$ref": "#/definitions/ContainerServiceVMSize",
+          "description": "Size of agent VMs."
+        },
+        "osDiskSizeGB": {
+          "$ref": "#/definitions/ContainerServiceOSDisk",
+          "description": "OS Disk Size in GB to be used to specify the disk size for every machine in this master/agent pool. If you specify 0, it will apply the default osDisk size according to the vmSize specified."
+        },
+        "vnetSubnetID": {
+          "$ref": "#/definitions/ContainerServiceVnetSubnetID",
+          "description": "VNet SubnetID specifies the vnet's subnet identifier."
+        },
+        "maxPods": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Maximum number of pods that can run on a node."
+        },
+        "osType": {
+          "$ref": "#/definitions/OSType",
+          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+        },
+        "maxCount":{
+          "type": "integer",
+          "format": "int32",
+          "description": "Maximun number of nodes for auto-scaling"
+        },
+        "minCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Minimun number of nodes for auto-scaling"
+        },
+        "enableAutoScaling":{
+          "type": "boolean",
```

Comment:
I see you've used a Boolean for this field - we try to encourage teams to use string/enums instead of Boolean as they are more descriptive and allow for non-breaking changes if you later wish to add more states.
The use of a Boolean here is actually [generating an ARM warning in the CI](https://github.com/Azure/azure-rest-api-specs/pull/4237/checks?check_run_id=24725712#user-content-ARM-Warnings) - is it possible to change this or is the API now live?

If not - then I wont block on it, but bare in mind for future API versions :)

Reply:
It only has 2 states, so boolean type is enough for the case.